### PR TITLE
feat: upgrade dependencies (swc_ecma_parser 0.122.19)

### PR DESCRIPTION
**Pull request by dsherret**
Need to test this out and open downstream prs.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -33,36 +33,36 @@ visit = ["swc_ecma_visit"]
...
-base64 = { version = "0.13.0", optional = true }
-data-url = { version = "0.1.1", optional = true }
-dprint-swc-ext = "0.4.0"
+base64 = { version = "0.13.1", optional = true }
+data-url = { version = "0.2.0", optional = true }
+dprint-swc-ext = "0.5.0"
 serde = { version = "1.0.144", features = ["derive"] }
 text_lines = { version = "0.6.0", features = ["serialization"] }
-url = { version = "2.2.2", features = ["serde"], optional = true }
+url = { version = "2.3.1", features = ["serde"], optional = true }
```
Comment (dsherret) [inline Cargo.toml:41]: @bartlomieju do you think upgrading url is ok?

Comment (bartlomieju) [reply Cargo.toml:41]: Yeah, I see no problem with it

**Review by bartlomieju — APPROVED** [review]

**Review by kitsonk — APPROVED** [review]
LGTM

One thing longer term is how we checked for "compatibility" without having to hope there are no regressions in the downstream crates as well as we test for features here.  For example TS4.9 introduces new `satisfies` operator, which appears to be supported in swc_ecma_parser `0.121.0` or later, but it is really hard to "know" and we wouldn't really be able to know until it is integrated all the way up the stack.

**Comment by dsherret** [comment]
@kitsonk we have these scripts (https://github.com/denoland/deno_ast/tree/main/scripts) for that.

I reported two bugs to swc yesterday after running them:

* https://github.com/swc-project/swc/issues/6239 (maybe a question and not a bug)
* https://github.com/swc-project/swc/issues/6238

**Pull request merged**